common/fileformat.cpp

```diff
@@ -0,0 +1,221 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/fileformat.h"
+
+#include "common/translation.h"
+#include "common/util.h"
+
+#include "image/bmp.h"
+#include "image/png.h"
+
+namespace Common {
+
+namespace FileFormat {
+
+struct AltMimeTypes {
+	const char *_mimeType;
+	Format _format;
+};
+
+AltMimeTypes g_altMimeTypes[] = {
+	{ "image/pict", kPICT, },
+};
+
+struct FormatTypeInfo {
+	const char *_mimeType;			///< MIME type
+	const char *_shortExtension;	///< Extension limited to 3 characters
+	const char *_longExtension;     ///< Extension with no length limit
+};
+
+const FormatTypeInfo g_formatTypeInfo[kCount] = {
+	{nullptr},
+
+	// Image formats
+	{"image/bmp", "bmp"},
+	{"image/png", "png"},
+	{"image/jpeg", "jpg"},
+	{"image/x-pict", "pct", "pict"},
+
+	// Audio formats
+	{"audio/x-wav", "wav"},
+	{"audio/x-aiff", "aif", "aiff"},
+
+	// Binary formats
+	{"application/octet-stream", nullptr},
+};
+
+ImageFormatCharacteristics::ImageFormatCharacteristics()
+	: _lossiness(kLossinessUnknown), _supportAlpha(false), _supportPalette(false), _maxBitDepth(0) {
+}
+
+AudioFormatCharacteristics::AudioFormatCharacteristics() : _lossiness(kLossinessUnknown) {
+}
+
+ImageSaveProperties::ImageSaveProperties() : _qualityLevel(75) {
+}
+
+const char *toMimeType(Format format) {
+	return g_formatTypeInfo[format]._mimeType;
+}
+
+Format fromMimeType(const Common::String &mimeType) {
```

Comment:
What is the usage of these mime types etc?

Reply:
Android intents will want MIME type

Reply:
As I don't plan to add this support to Android, maybe iOS will need this?
Or maybe @antoniou79 will want to implement it?